src/Stryker.Core/Stryker.Core/TestRunners/VsTest/VsTestRunner.cs

```diff
@@ -0,0 +1,212 @@
+
+            waitHandle.WaitOne();
+
+            return handler.DiscoveredTestCases;
+        }
+
+        private IEnumerable<TestResult> RunSelectedTests(IEnumerable<TestCase> testCases, string runSettings)
+        {
+            var waitHandle = new AutoResetEvent(false);
+            var handler = new RunEventHandler(waitHandle, _messages);
+            _vsTestConsole.RunTests(testCases, runSettings, handler);
+
+            waitHandle.WaitOne();
+            return handler.TestResults;
+        }
+
+        private IEnumerable<TestResult> RunAllTests(int? activeMutationId, string runSettings)
+        {
+            var runCompleteSignal = new AutoResetEvent(false);
+            var processExitedSignal = new AutoResetEvent(false);
+            var handler = new RunEventHandler(runCompleteSignal, _messages);
+            var testHostLauncher = new StrykerVsTestHostLauncher(() => processExitedSignal.Set(), activeMutationId);
+
+            _vsTestConsole.RunTestsWithCustomTestHost(_sources, runSettings, handler, testHostLauncher);
+
+            // Test host exited signal comes after the run complete
+            processExitedSignal.WaitOne();
+
+            // At this point, run must have complete. Check signal for true
+            runCompleteSignal.WaitOne();
+
+            return handler.TestResults;
+        }
+
+        private TraceLevel DetermineTraceLevel()
+        {
+            switch (_options.LogOptions.LogLevel)
+            {
+                case LogEventLevel.Debug:
+                case LogEventLevel.Verbose:
+                    return TraceLevel.Verbose;
+                case LogEventLevel.Error:
+                    return TraceLevel.Error;
+                case LogEventLevel.Warning:
+                    return TraceLevel.Warning;
+                case LogEventLevel.Information:
+                    return TraceLevel.Info;
+                default:
+                    return TraceLevel.Off;
+            }
+        }
+
+        private string GenerateRunSettings(int timeout)
+        {
+            string targetFramework = _projectInfo.TargetFramework;
+
+            string targetFrameworkVersion = Regex.Replace(targetFramework, @"[^.\d]", "");
+            switch (targetFramework)
+            {
+                case string s when s.Contains("netcoreapp"):
+                    targetFrameworkVersion = $".NETCoreApp,Version = v{targetFrameworkVersion}";
+                    break;
+                case string s when s.Contains("netstandard"):
+                    throw new Exception("Unsupported targetframework detected. A unit test project cannot be netstandard!: " + targetFramework);
+                default:
+                    targetFrameworkVersion = $".NETFramework = v{targetFrameworkVersion}";
+                    break;
+            }
+
+            return $@"<RunSettings>
+  <RunConfiguration>
+    <MaxCpuCount>1</MaxCpuCount>
```

Comment:
Why is this set to 1? We might get even more performance if we set this higher?

Reply:
I set the max cpu count to 1 because the testruns are already distributed over multiple threads with different test runners by stryker. I don't believe it would be beneficial to performance if vstest is then also allowed to spawn even more threads.